src/Riok.Mapperly/Descriptors/Mappings/ForEachAddEnumerableMapping.cs

```diff
@@ -14,24 +14,36 @@ namespace Riok.Mapperly.Descriptors.Mappings;
 public class ForEachAddEnumerableMapping : ExistingTargetMappingMethodWrapper
 {
     private readonly ObjectFactory? _objectFactory;
+    private readonly ITypeSymbol _typeToInstantiate;
+    private readonly string? _countPropertyName;
 
     public ForEachAddEnumerableMapping(
         ITypeSymbol sourceType,
         ITypeSymbol targetType,
         ITypeMapping elementMapping,
         ObjectFactory? objectFactory,
         string insertMethodName,
-        EnsureCapacity? ensureCapacityBuilder
+        EnsureCapacity? ensureCapacityBuilder,
+        ITypeSymbol? typeToInstantiate = null,
+        string? countPropertyName = null
```

Comment:
Add a comment explaining that this property can only be set if `typeToInstantiate` has a constructor with exact one int parameter. If this parameter is set, the ensure capacity isn't needed, is it?

Reply:
Should I create a separate constructor? `ForEachSetDictionaryMapping` probably needs one as well.

Reply:
A separate constructor may be a good solution.